kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/main/java/org/kie/server/remote/rest/common/util/RestUtils.java

```diff
@@ -264,4 +268,24 @@ public static String errorMessage(Throwable e, String defaultMessage) {
     public static String errorMessage(Throwable e) {        
         return errorMessage(e, MessageFormat.format(UNEXPECTED_ERROR, e.getMessage()));
     }
+    
+    public static Boolean isCorrelationKeyAlreadyExists(Exception e) {
+        SQLIntegrityConstraintViolationException constraintViolationException = null;
+
+        Throwable t = e.getCause();
+        while ((t != null) && !(t instanceof SQLIntegrityConstraintViolationException))
+            t = t.getCause();
+
+        constraintViolationException = (SQLIntegrityConstraintViolationException) t;
+
+        /*
+         * 23000 SQLState is used by MySQL, MSSQL,Oracle,Postgres and MariaDB for constraint Violation. 
+         * 'already exists' keyword is used in RuntimeException which is thrown by org.jbpm.persistence.JpaProcessPersistenceContext class to indicate that Correlation key already exists in database. 
+         * */
+        if ((constraintViolationException != null && (constraintViolationException.getSQLState().equals("23000"))) || e.getMessage().contains("already exists")) {
```

Comment:
Could be written the other way round: 
```suggestion
        if ("23000".equals(constraintViolationException.getSQLState()) || e.getMessage().contains("already exists")) {
```

Reply:
@gmunozfe  when we use H2 db root exception is org.h2.jdbc.JdbcSQLException not the java.sql.SQLIntegrityConstraintViolationException so if I modify if loop as per your comment, it throws NPE For H2 DB.

Reply:
OK, forget my comment then